backend/fms_core/viewsets/full_sample.py

```diff
@@ -0,0 +1,184 @@
+from collections import Counter
+from django.core import serializers
+import json
+
+from rest_framework import viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from django.db.models import Count, Q, F
+from django.core.exceptions import ValidationError
+
+from datetime import datetime
+from ..utils import RE_SEPARATOR, float_to_decimal
+
+
+from fms_core.models import FullSample, Biosample, Sample, DerivedSample, SampleKind, Container, Individual, DerivedBySample
+from fms_core.serializers import FullSampleSerializer, FullSampleExportSerializer, SampleSerializer
+from fms_core.template_importer.importers import SampleSubmissionImporter, SampleUpdateImporter
+
+from fms_core.template_paths import SAMPLE_SUBMISSION_TEMPLATE, SAMPLE_UPDATE_TEMPLATE
+
+from ._constants import _full_sample_filterset_fields
+from ._utils import TemplateActionsMixin, _list_keys, versions_detail
+from fms_core.services.sample import create_full_sample
+
+class FullSampleViewSet(viewsets.ModelViewSet, TemplateActionsMixin):
+    queryset = FullSample.objects.select_related("individual", "container", "sample_kind", "biosample", "derived_sample").all()
```

Comment:
biosample and derived_sample should not be part of the select_related. You should have all their content already in full sample. The only use of those FK are the id.